packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ServiceVersionDefinition.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Generator.CSharp.ClientModel.Primitives;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Providers;
+using static Microsoft.Generator.CSharp.Snippets.Snippet;
+
+namespace Microsoft.Generator.CSharp.ClientModel.Providers
+{
+    internal class ServiceVersionDefinition : TypeProvider
+    {
+        private readonly IReadOnlyList<ApiVersion> _apiVersions;
+        private IReadOnlyList<EnumTypeMember>? _members;
+
+        public ServiceVersionDefinition(ClientOptionsProvider declaringType)
```

Comment:
any reason we can't use fixedenumprovider?

Reply:
That provider takes in an InputEnumType in the ctor. Are you suggesting I convert the api version string to an InputEnumType ?

Reply:
I think we cannot reuse `FixedEnumProvider` because it is internal in MGC project.
We could make it public and inherit it here because we want to add a declaring type provider here, and I think it does not make much sense we only change the definition of FixedEnumProvider to let it take a DeclaringTypeProvider but leave others unchanged

Reply:
@ArcturusZhang @m-nash - Let me know if we think it's worth finding a way to re-use FixedEnumProvider. Though all the point here are valid, and I think we would need to make it public and possibly add some additional logic or override to handle the naming of the members for api versions.